### PR TITLE
Separate docs source from parsing v2

### PR DESCRIPTION
**Pull request by iwahbe**
This PR refactors the code used to acquire the markdown files we source our docs from. The goal is to make it easy to test file acquisition and parsing separately.

The new interface looks like this:
https://github.com/pulumi/pulumi-terraform-bridge/blob/856055a0cd3238780c9c35bbe605282f75cc2de7/pkg/tfgen/source.go#L30-L47

This PR is best reviewed as two separate commits:
- d46360529f7198cf9f3f37df3756e3308726e10e is a byte for byte copy of the functionality that we want to refactor from `docs.go` to the newly created `source.go`.
- 856055a0cd3238780c9c35bbe605282f75cc2de7 implements and consumes the interface. 

This PR does not have *any* effect on downstream users. It is purely a refactor for maintainability.

### CODE REVIEW

File: pkg/tfgen/source.go [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+	"github.com/pulumi/pulumi/sdk/v3/go/common/diag"
+
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge"
+)
+
+// A source of documentation bytes.
+type DocsSource interface {
+	// Get the bytes for a resource with TF token rawname.
+	getResource(
+		rawname string, info tfbridge.ResourceOrDataSourceInfo,
```
Comment (t0yv0) [inline pkg/tfgen/source.go:34]: Repeating the comment before, the signature is very permissive, does it really need to know tfbridge.ResourceOrDataSourceInfo and why?

**Review by t0yv0 — APPROVED** [review]
Love pure refactoring PRs, easy approve. Directionally this is very good we need to untangle this code here gradually.

**Review by mjeffryes — APPROVED** [review]
I'll also echo my comment from the previous version that it would be nice to add some units on the new interfaces, but the refactor is straightforward here and seems like an improvement overall.

File: pkg/tfgen/source.go (hunk 1 above)
Comment (iwahbe) [reply pkg/tfgen/source.go]: I reduced `info` to `*tfbridge.DocInfo`, which it really does need.

**Comment by iwahbe** [comment]
> I'll also echo my comment from the previous version that it would be nice to add some units on the new interfaces, but the refactor is straightforward here and seems like an improvement overall.

I'll merge unit tests with https://github.com/pulumi/pulumi-terraform-bridge/pull/1388.

**Pull request merged**